include/dxc/HLSL/DxilSignatureAllocator.inl

```diff
@@ -332,117 +348,152 @@ unsigned DxilSignatureAllocator::PackOptimized(std::vector<PackElement*> element
-    if (!SE->IsAllocated()) {
-      continue;
-    }
-    unsigned row = SE->GetStartRow();
-    DXASSERT_NOMSG(row < clipcullRegUsed);
-    clipcullElementsByRow[row].push_back(SE);
-    clipcullComponentsByRow[row] += SE->GetCols();
-    // Deallocate element, to be allocated later:
-    SE->ClearLocation();
-  }
-  // Init temp elements, used to find compatible spaces for subsets:
-  DummyElement clipcullTempElements[2];
-  for (unsigned row = 0; row < clipcullRegUsed; ++row) {
-    DXASSERT_NOMSG(!clipcullElementsByRow[row].empty());
-    clipcullTempElements[row].kind = clipcullElementsByRow[row][0]->GetKind();
-    clipcullTempElements[row].interpolation = clipcullElementsByRow[row][0]->GetInterpolationMode();
-    clipcullTempElements[row].interpretation = clipcullElementsByRow[row][0]->GetInterpretation();
-    clipcullTempElements[row].dataBitWidth = clipcullElementsByRow[row][0]->GetDataBitWidth();
-    clipcullTempElements[row].rows = 1;
-    clipcullTempElements[row].cols = clipcullComponentsByRow[row];
-  }
-
   // ==========
   // Allocate 4-component elements
   if (!vec4Elements.empty()) {
     std::sort(vec4Elements.begin(), vec4Elements.end(), CmpElementsLess);
-    unsigned used = PackGreedy(vec4Elements, startRow, numRows);
-    startRow += used;
-    numRows -= used;
-    if (rowsUsed < used)
-      rowsUsed = used;
+    rowsUsed = std::max(rowsUsed, PackGreedy(vec4Elements, startRow, numRows));
+    startRow = std::max(startRow, rowsUsed);
   }
 
   // ==========
   // Allocate indexed tessfactors in rightmost column
   if (!indexedtessElements.empty()) {
     std::sort(indexedtessElements.begin(), indexedtessElements.end(), CmpElementsLess);
-    unsigned used = PackGreedy(indexedtessElements, startRow, numRows, 3);
-    if (rowsUsed < used)
-      rowsUsed = used;
+    rowsUsed = std::max(rowsUsed, PackGreedy(indexedtessElements, startRow, numRows, 3));
   }
 
   // ==========
   // Allocate arbitrary
   if (!arbElements.empty()) {
     std::sort(arbElements.begin(), arbElements.end(), CmpElementsLess);
-    unsigned used = PackGreedy(arbElements, startRow, numRows);
-    if (rowsUsed < used)
-      rowsUsed = used;
+    rowsUsed = std::max(rowsUsed, PackGreedy(arbElements, startRow, numRows));
   }
 
   // ==========
   // Allocate system values
   if (!svElements.empty()) {
     std::sort(svElements.begin(), svElements.end(), CmpElementsLess);
-    unsigned used = PackGreedy(svElements, startRow, numRows);
-    if (rowsUsed < used)
-      rowsUsed = used;
+    rowsUsed = std::max(rowsUsed, PackGreedy(svElements, startRow, numRows));
   }
 
   // ==========
   // Allocate clip/cull
-  for (unsigned i = 0; i < clipcullRegUsed; ++i) {
-    bool bAllocated = false;
-    unsigned cols = clipcullComponentsByRow[i];
-    for (unsigned row = startRow; row < startRow + numRows; ++row) {
-      if (DetectRowConflict(&clipcullTempElements[i], row))
+  std::sort(clipcullElements.begin(), clipcullElements.end(), CmpElementsLess);
```

Comment:
With this code having moved, does the "// Packing overview" comment higher up need updating?

Reply:
Updating..